### PR TITLE
Drop clj-kondo dir from linting

### LINKED ISSUE #17947: check in taoensso and babashka inside `./clj-kondo` into version control but prevent them from being formatted
ref : https://github.com/status-im/status-mobile/pull/17783 

The PR above simply adds  these locations to `.gitignore` 

- `.clj-kondo/taoensso/`
- `.clj-kondo/babashka/`

However as mentioned : https://clojurians-log.clojureverse.org/clj-kondo/2022-03-18 they _**ideally**_  belong in the source control.
Unless the `lsp` configurations are modified to ignore these paths these files will get auto formatted every time the command `make lint-fix` is run.
The formatting varies on `macos` and `linux` systems.

The ideal solution would be to remove those entries from `.gitingore` and modify the `lsp` configurations in such a way that those folders are not formatted by the linter.

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17947 

### Summary
Adjusts the clojure file grep search pattern to not include files in ./clj-kondo when running `make lint-fix` or `make lint`

status: ready

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -305,7 +305,7 @@ endif
 
 # Get all clojure files, including untracked, excluding removed
 define find_all_clojure_files
-$$(comm -23 <(sort <(git ls-files --cached --others --exclude-standard)) <(sort <(git ls-files --deleted)) | grep -e \.clj$$ -e \.cljs$$ -e \.cljc$$ -e \.edn)
+$$(comm -23 <(sort <(git ls-files --cached --others --exclude-standard)) <(sort <(git ls-files --deleted)) | grep -e '^(?!clj-kondo).*\.clj'$$ -e \.cljs$$ -e \.cljc$$ -e '^(?!clj-kondo).*\.clj')
```
Comment (siddarthkay) [inline Makefile]: I'm not 100% sold that this will work. I'll do a few tests and get back to you.

Comment (ibrkhalil) [reply Makefile]: Sorry I had a typo, Pushed a fix.

Comment (siddarthkay) [reply Makefile]: nice!

**Review by siddarthkay — APPROVED** [review]
Looks good to me, I would also like @ilmotta and @yqrashawn to have a look since they have context.

**Review by ilmotta — COMMENTED** [review]
Thanks @ibrkhalil.

We shouldn't ignore `.clj-kondo/` entirely. The rule should be:

Only lint files under `.clj-kondo/status-im/` and `.clj-kondo/config.edn`, the rest is assumed to be auto-generated from our dependencies and are out of our control.

**Comment by ibrkhalil** [comment]
> Thanks @ibrkhalil.
> 
> We shouldn't ignore `.clj-kondo/` entirely. The rule should be:
> 
> Only lint files under `.clj-kondo/status-im/` and `.clj-kondo/config.edn`, the rest is assumed to be auto-generated from our dependencies and are out of our control.

Improved, Can you please take another look?

**Review by ilmotta — APPROVED** [review]
LGTM, thank you @ibrkhalil!

**Review by yqrashawn — APPROVED** [review]

**Pull request merged**